### PR TITLE
feat: navigation menu added, header visual update applied

### PR DESCRIPTION
**Pull request by AnatoliiKim**

### CODE REVIEW

File: src/app/shared/components/main-nav/main-nav.component.scss [hunk 1]
```diff
@@ -0,0 +1,62 @@
+@import 'src/app/shared/styles/variables';
+
+:host {
+  ::ng-deep {
```
Comment (vyMoon) [inline src/app/shared/components/main-nav/main-nav.component.scss:4]: can we avoid ng deep?

File: src/app/auth/models/role.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import { UserRole } from '@app/shared/constants/user-role.enum';
+
 export type Role = {
   id: number;
   name: string;
-  slug: string;
+  slug: UserRole;
```
[the remark this answers is not in the document]
Comment (AnatoliiKim) [reply src/app/auth/models/role.ts:6]: do you think to use name property instead?

File: src/app/shared/constants/nav-menu-role-mapping.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  {
+    name: 'Профиль',
+    route: 'profile',
+  },
+  {
+    name: 'Мои заявки',
+    route: 'profile/requests',
+  },
+];
```
[the remark this answers is not in the document]
Comment (AnatoliiKim) [reply src/app/shared/constants/nav-menu-role-mapping.ts:32]: Yes, already fixed it in the next commit

File: src/app/shared/components/main-nav/main-nav.component.scss (hunk 1 above)
Comment (AnatoliiKim) [reply src/app/shared/components/main-nav/main-nav.component.scss:4]: I had to use ng-deep to customize side-nav material styles

File: src/app/shared/components/main-nav/main-nav.component.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export class MainNavComponent implements OnInit {
+  links: NavigationLink[] = [];
+  title = this.mainPageHeaderService.getPageTitle();
+
+  constructor(private readonly mainPageHeaderService: MainPageHeaderService, private readonly store: Store) {}
+
+  ngOnInit(): void {
+    this.links = <NavigationLink[]>(
+      navLinksMap.get(this.store.selectSnapshot(AuthState.user)?.role.slug ?? UserRole.default)
+    );
```
[the remark this answers is not in the document]
Comment (AnatoliiKim) [reply src/app/shared/components/main-nav/main-nav.component.ts:24]: fixed using select decorator

File: src/app/auth/models/role.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnatoliiKim) [reply src/app/auth/models/role.ts]: slug property deleted

File: src/app/shared/components/main-nav/main-nav.component.scss (hunk 1 above)
Comment (AnatoliiKim) [reply src/app/shared/components/main-nav/main-nav.component.scss]: ng-deep removed, I used own wrapper instead

File: src/app/shared/components/main-nav/main-nav.component.html [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      [mode]="'over'"
+      [opened]="false">
+    <div class="nav-wrapper">
+      <mat-toolbar class="nav-head">
+        <button type="button" class="close" mat-icon-button (click)="drawer.toggle()">
+          <mat-icon class="close-icon">close</mat-icon>
+        </button>
+      </mat-toolbar>
+      <mat-nav-list class="nav-list">
+        <a *ngFor="let link of links" class="nav-link" mat-list-item [routerLink]="link.route" [routerLinkActive]="['active']" (click)="drawer.toggle()">{{link.name}}</a>
```
Comment (vyMoon) [inline src/app/shared/components/main-nav/main-nav.component.html:14]: as I can see we should have s separator between user links and admin links. Do we have one?

**Review by vyMoon — APPROVED** [review]

**Pull request merged**